lib/plugins/aws/deploy/compile/events/s3/index.js

```diff
@@ -38,6 +39,43 @@ class AwsCompileS3Events {
               if (event.s3.event) {
                 notificationEvent = event.s3.event;
               }
+              if (event.s3.rules) {
+                if (!event.s3.event) {
```

Comment:
yeah so the `notificationEvent` should always be ObjectCreated until the user overwrite that. So I guess as far as your code is concerned, if it simply runs later, it could simply use that `notificationEvent` variable